canal.cc

```diff
@@ -300,11 +346,16 @@ mainExcept(int argc, char *argv[])
                 }
             }
         } else {
-            for (auto loc = hdr.p_vaddr; loc < hdr.p_vaddr + hdr.p_filesz; loc += sizeof p) {
+            const size_t step = sizeof(Elf::Off);
+            data.resize(hdr.p_filesz/step);
```

Comment:
... here's where you define data as std::vector<Elf::Off>data(size);

Also, in a core, an individual section could be enormous, so you should cap the size of data to something like a few pages, and iterate over that size in an outer loop. Otherwise, the memory footprint for canal could go from a few megabytes to a few gig if a process decides to use a heap management strategy that maps in massive chunks of anon memory. (and the system you run canal on may not have the same resources as the machine that generated the core)